tests/components/prometheus/test_init.py

```diff
@@ -1113,6 +856,485 @@ async def test_disabling_entity(hass, hass_client):
     )
 
 
+def simulate_entities(
```

Comment:
I'd create a pytest fixture for each entity domain we want to test a state for instead of this common method. The tests can then mix and match fixtures as they need.

Reply:
I think fixtures are loaded before the test function is executed? Since the prometheus integration is listening to `state_changed` and `entity_registry_updated` events and ` setup_prometheus_client` is called after that, these events would be missed.

Should `setup_prometheus_client()` also be fixture?

Reply:
Yes, make a fixture of that and a separate fixture for the namespace. We can override the fixture namespace, via `pytest.mark.parametrize` in the test that needs a different namespace.

Reply:
I added a new commit with the fixtures.